graph/db/sql_store.go

```diff
@@ -43,3 +99,768 @@ func NewSQLStore(db BatchedSQLQueries, kvStore *KVStore) *SQLStore {
 		KVStore: kvStore,
 	}
 }
+
+// TxOptions defines the set of db txn options the SQLQueries
+// understands.
+type TxOptions struct {
+	// readOnly governs if a read only transaction is needed or not.
+	readOnly bool
+}
+
+// ReadOnly returns true if the transaction should be read only.
+//
+// NOTE: This implements the TxOptions.
+func (a *TxOptions) ReadOnly() bool {
+	return a.readOnly
+}
+
+// NewReadTx creates a new read transaction option set.
+func NewReadTx() TxOptions {
+	return TxOptions{
+		readOnly: true,
+	}
+}
+
+// AddLightningNode adds a vertex/node to the graph database. If the node is not
+// in the database from before, this will add a new, unconnected one to the
+// graph. If it is present from before, this will update that node's
+// information.
+//
+// NOTE: part of the V1Store interface.
+func (s *SQLStore) AddLightningNode(node *models.LightningNode,
```

Comment:
currently working on a benchmark to compare this to the performance of the kvdb version which uses the batch schedular. We may end up needing something similar here to batch these txs